lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -62,96 +62,84 @@ function dosomething_northstar_menu() {
 function dosomething_northstar_create_user($user, $password = null) {
-  $client = _dosomething_northstar_build_http_client();
   $payload = dosomething_northstar_transform_user($user, $password);
 
-  $response = drupal_http_request($client['base_url'] . '/users', [
-    'headers' => $client['headers'],
-    'method' => 'POST',
-    'data' => json_encode($payload),
-  ]);
+  try {
+    $response = dosomething_northstar_client()->post('v1/users', $payload);
+  }
+  catch (Exception $error) {
+    watchdog('dosomething_northstar', 'User NOT migrated to Northstar: ' . $error->getMessage(), NULL, WATCHDOG_ERROR);
+
+    return NULL;
+  }
 
   // Save the newly registered user's ID to their local profile field.
-  dosomething_northstar_save_id_field($user->uid, json_decode($response->data));
+  dosomething_northstar_save_id_field($user->uid, $response);
 
   // Add to request log if enabled.
   dosomething_northstar_log_request('register_user', $user, $payload, $response);
 
-  if (! _dosomething_northstar_is_successful_response($response)) {
-    watchdog('dosomething_northstar', 'User not migrated : ' . $response->code, NULL, WATCHDOG_ERROR);
-    return;
-  }
-
   if (module_exists('stathat')) {
     stathat_send_ez_count('drupal - Northstar - user migrated - count', 1);
   }
+
+  return $response;
 }
 
 /**
- * Update a user's existing Northstar profile based on
- * their Drupal ID.
+ * Update a user's existing Northstar profile based on their Drupal ID.
  *
- * @param $user - Drupal user object
- * @param $password - Unhashed password provided during registration.
+ * @param object $user - Drupal user object
+ * @param string $password - Unhashed password provided during registration.
  */
 function dosomething_northstar_update_user($user, $password = null) {
-  $client = _dosomething_northstar_build_http_client();
   $payload = dosomething_northstar_transform_user($user, $password);
 
-  $response = drupal_http_request($client['base_url'] . '/users/drupal_id/' . $user->uid, [
-    'headers' => $client['headers'],
-    'method' => 'PUT',
-    'data' => json_encode($payload),
-  ]);
+  try {
+    $response = dosomething_northstar_client()->put('v1/users/drupal_id/' . $user->uid, $payload);
+  }
+  catch (Exception $error) {
+    watchdog('dosomething_northstar', 'User NOT updated on Northstar: ' . $error->getMessage(), NULL, WATCHDOG_ERROR);
+
+    return NULL;
+  }
 
   // Add to request log if enabled.
   dosomething_northstar_log_request('update_user', $user, $payload, $response);
 
-  if (_dosomething_northstar_is_successful_response($response)) {
-    if (module_exists('stathat')) {
-      stathat_send_ez_count('drupal - Northstar - user updated - count', 1);
-    }
-  }
-  elseif ($response->code === '404') {
-    // If the given Drupal ID 404s, try to register them.
-    $password_or_null = ! empty($form_state['values']['pass']) ? $form_state['values']['pass'] : null;
-    dosomething_northstar_create_user($user, $password_or_null);
-  }
```

Comment:
Oh yep, that certainly seems like a mistake. Should have been using the `$password` parameter.

404s will return null (see [`send()`](https://github.com/DoSomething/gateway/blob/2de6ade1b724fc93353545996426ecee49b7cd7e/src/Common/RestApiClient.php#L216-L240) in RestApiClient) so checking for null seems fine to me. Other types of error codes should trigger their respective exceptions.